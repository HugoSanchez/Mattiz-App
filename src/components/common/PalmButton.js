import React from 'react';
import { Text, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Feather';

import colors from '../../constants/colors';

// Props:
// @backgroundColor: -Optional- Wrapping container color.
// @iconName:        -Required- Icon to dislplay, if any.
// @onPress:         -Required- Function to be executed on press. 
// @title:           -Required- Text to be displayed. 

const PalmButton = props => {
    const { container, titleStyle } = styles;
    return (
        <TouchableOpacity
            onPress={ props.onPress } 
            style={[ container, props.backgroundColor ? { backgroundColor: props.backgroundColor} : null ]}>
            <Text 
                style={[ titleStyle, props.titleStyle ]}> 
                {
                    props.title
                } 
            </Text>
            {
                props.iconName ?
                <Icon 
                    name={ props.iconName } 
                    size={ 27 } 
                    color={ colors.paletteDarkBlue }
                    style={{ marginLeft: 5 }} 
                />
                :
                null 
            }
        </TouchableOpacity>
    );
}

const styles = {
    container: {
        height: 55,
        margin: 10, 
        borderRadius: 5,
        alignItems: 'center',
        justifyContent: 'center',
        flexDirection: 'row',
        backgroundColor: 'transparent',
        shadowColor: '#000',
        shadowOffset: { width: 2, height: 5 },
        shadowOpacity: 0.29,
        shadowRadius: 4.65,
        elevation: 12,
    },
    titleStyle: { 
        fontFamily: 'Rajdhani-Regular', 
        fontSize: 22, 
        color: colors.palleteDarkGreen
    }
}

export default PalmButton;