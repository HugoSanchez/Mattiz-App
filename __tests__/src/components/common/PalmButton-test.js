import 'react-native'
import React from 'react'
import PalmButton from '../../../../src/components/common/PalmButton'
import {render, fireEvent} from '@testing-library/react-native'

test('calls onPress callback when clicked', () => {
	const palmButtonCallback = jest.fn()
	const {container} = render(<PalmButton onPress={palmButtonCallback} />)

	fireEvent.press(container.children[0])

	expect(palmButtonCallback).toHaveBeenCalled()
	expect(container.children[0]).toMatchSnapshot()
})

test('Renders with title', () => {
	const {container} = render(<PalmButton title="Button" />)
	expect(container.children[0]).toMatchSnapshot()
})
