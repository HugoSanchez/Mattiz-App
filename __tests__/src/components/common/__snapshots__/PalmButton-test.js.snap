// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders with title 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 5,
        "elevation": 12,
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "center",
        "margin": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 5,
          "width": 2,
        },
        "shadowOpacity": 0.29,
        "shadowRadius": 4.65,
      },
      null,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#154053",
          "fontFamily": "Rajdhani-Regular",
          "fontSize": 22,
        },
        undefined,
      ]
    }
  >
    Button
  </Text>
</TouchableOpacity>
`;

exports[`calls onPress callback when clicked 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "transparent",
        "borderRadius": 5,
        "elevation": 12,
        "flexDirection": "row",
        "height": 55,
        "justifyContent": "center",
        "margin": 10,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 5,
          "width": 2,
        },
        "shadowOpacity": 0.29,
        "shadowRadius": 4.65,
      },
      null,
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#154053",
          "fontFamily": "Rajdhani-Regular",
          "fontSize": 22,
        },
        undefined,
      ]
    }
  />
</TouchableOpacity>
`;
